src/chatd.cpp

```diff
@@ -4315,11 +4471,116 @@ void FilteredHistory::clear()
     init();
 }
 
+HistSource FilteredHistory::loadHistory(uint32_t count)
+{
+    uint32_t messagesLoaded = 0;
+    if ((mOldestNotifyMsg != --mBuffer.end() && mBuffer.size() > 1) || mFirstNotification)
```

Comment:
I don't understand this checkup. First of all, we should notify messages from RAM if there are messages after `mOldestNotifyMsg`, even if there's only one message, right?
Second, the `mFirstNotification` seems to be useless, since it's reset in all the code-paths here. As long as there are messages available and not-yet-notified in RAM, we should go for that `HistSource`.
Please, let me know if I'm wrong.